configure.ac

```diff
@@ -7007,9 +7007,10 @@ int main()
     }
     return 0; // all good
 }
-]])],[
-  ac_cv_libatomic_needed=no
-],[ac_cv_libatomic_needed=yes],[ac_cv_libatomic_needed=yes])
+]])],
+  [ac_cv_libatomic_needed=no],  dnl build succeeded
+  [ac_cv_libatomic_needed=yes], dnl build failed
+  [ac_cv_libatomic_needed=no])  dnl cross compilation
```

Comment:
We could have used `ac_cv_libatomic_needed=n/a` for a more accurate result message.

Reply:
Oh, I didn't know this value.

Reply:
> Oh, I didn't know this value.

Actually, the value is up to you; it's not a magical GNU Autoconf value :) See docs for [AC_CACHE_CHECK](https://www.gnu.org/software/autoconf/manual/autoconf-2.71/html_node/Caching-Results.html). The signature for that macro is:

```
AC_CACHE_CHECK (message, cache-id, commands-to-set-it)
```

Quoting the docs:

> It calls `AC_MSG_CHECKING` for message, then `AC_CACHE_VAL` with the *cache-id* and commands arguments, and `AC_MSG_RESULT` with *cache-id*.

In out case, _cache-id_ is `ac_cv_libatomic_needed`. Whatever we set it to will be displayed to the user at the end of the check (via the implicit `AC_MSG_RESULT` call). So if we set `ac_cv_libatomic_needed` to `n/a`, the user should see this when cross-compiling:

> checking whether libatomic is needed by <pyatomic.h>... n/a

Currently, the user will see this when cross-compiling:

> checking whether libatomic is needed by <pyatomic.h>... no

The "no" may lead the user to incorrectly assume that `configure` actually checked whether libatomic was needed (and that `configure` concluded it was not).

I think it may be worth it to adjust this message. What do you think?

Reply:
Oh, my suggestion is made moot by #109344